### PR TITLE
Jit_LoadStorePaired: Make psq_st PIE-compliant

### PR DESCRIPTION
**Pull request by MerryMage**
Uses 8 bit operations on a 64 bit register. Aligns jump-table to 256 bytes for this to work.

Alternative considered: Use RSCRATCH2 then rematerialize RSCRATCH2:

    MOV(32, R(RSCRATCH2), Imm32(0x7));
    AND(32, R(RSCRATCH2), PPCSTATE(spr[SPR_GQR0 + i]));
    LEA(64, RSCRATCH, M(w ? asm_routines.singleStoreQuantized : asm_routines.pairedStoreQuantized));
    LEA(64, RSCRATCH, MComplex(RSCRATCH, RSCRATCH2, SCALE_8, 0));

    MOV(32, R(RSCRATCH2), Imm32(0x3F07));
    AND(32, R(RSCRATCH2), PPCSTATE(spr[SPR_GQR0 + i]));

    CALLptr(MatR(RSCRATCH));

Which is probably cleaner, though larger in codesize.

### CODE REVIEW

File: Source/Core/Common/x64Emitter.cpp [hunk 1]
```diff
@@ -120,6 +120,16 @@ void XEmitter::ReserveCodeSpace(int bytes)
     *code++ = 0xCC;
 }
 
+const u8* XEmitter::AlignCodeTo(size_t alignment)
```
Comment (lioncash) [inline Source/Core/Common/x64Emitter.cpp]: would it be possible to implement the other alignment functions in terms of this one? (like AlignCode4, etc)

Comment (MerryMage) [inline Source/Core/Common/x64Emitter.cpp]: absolutely

File: Source/Core/Core/PowerPC/Jit64/Jit_LoadStorePaired.cpp [hunk 2]
```diff
@@ -89,12 +89,12 @@ void Jit64::psq_stXX(UGeckoInstruction inst)
...
-
-    if (w)
-      CALLptr(MScaled(RSCRATCH, SCALE_8, PtrOffset(asm_routines.singleStoreQuantized)));
-    else
-      CALLptr(MScaled(RSCRATCH, SCALE_8, PtrOffset(asm_routines.pairedStoreQuantized)));
+    LEA(64, RSCRATCH, M(w ? asm_routines.singleStoreQuantized : asm_routines.pairedStoreQuantized));
+    // 8-bit operations do not zero upper 32-bits of 64-bit registers.
+    // Here we know that RSCRATCH's least significant byte is zero.
+    OR(8, R(RSCRATCH), R(RSCRATCH2));
+    SHL(8, R(RSCRATCH), Imm8(3));
```
Comment (degasus) [inline Source/Core/Core/PowerPC/Jit64/Jit_LoadStorePaired.cpp]: Will this mask the input or output of the shift with 0xff ?

File: Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.cpp [hunk 3]
```diff
@@ -243,7 +243,7 @@ constexpr std::array<u8, 8> sizes{{32, 0, 0, 0, 8, 16, 8, 16}};
 
 void CommonAsmRoutines::GenQuantizedStores()
 {
-  pairedStoreQuantized = reinterpret_cast<const u8**>(const_cast<u8*>(AlignCode16()));
+  pairedStoreQuantized = reinterpret_cast<const u8**>(const_cast<u8*>(AlignCodeTo(256)));
```
Comment (degasus) [inline Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.cpp]: Maybe a small comment here why 256 byte alignment is a hard requirement

**Comment by degasus** [comment]
The assembly LGTM.

**Pull request merged**